### PR TITLE
Table: migrate old-table config to new table config

### LINKED ISSUE #23822: Migration button for old to new Table panel
Todo:

* [ ] Add migration button to old table editor 

Reference:
Singlstat editor
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/singlestat/editor.html#L9
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/singlestat/module.ts#L143

Old table edtor: 
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/singlestat/editor.html

Add a function to the old table angular controller that calls   this.onPluginTypeChange('table');

Add migration logic here: 
https://github.com/grafana/grafana/blob/c4d91e25966ee39213c7ed7cf06cdc263757b511/public/app/plugins/panel/table/migrations.ts#L29 

Use this old devenv test dashboard to test migration: 
http://localhost:3000/d/pttable/panel-tests-table?orgId=1 

There are a lot of tricky changes between old and new table panel. For example old panel had built-in transformations. Transformations are now an external core feature not linked to a specific panel. 

Old table transforms to new transform feature mapping
* Time series to columns => Outer join transform (but id is seriesToColumns) 

```json
"transformations": [
    {
      "id": "seriesToColumns",
      "options": {}
    }
  ]
```

* Series to rows 
```json
"transformations": [
    {
      "id": "seriesToRows",
      "options": {}
    }
  ],
```

the old table panel.columns should be mapped to new override rule system.

Similar to how this migration https://github.com/grafana/grafana/pull/29599/files added migration from seriesOverrides (in old graph) to new override rules.

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR adds a migration button to the old (angular) table panel and code to map old config to new table panel config when the button is clicked...

https://user-images.githubusercontent.com/73201/104027198-54530200-51c7-11eb-9543-b1feff87687f.mp4

**Which issue(s) this PR fixes**:

Fixes #23822

**Special notes for your reviewer**:

### CODE REVIEW

**Review by dprokop — COMMENTED** [review]
Took a quick glimpse on the code and so far it looks good. I will do some more testing during the day

**Comment by torkelo** [comment]
Works great. The only option that seems to not migrate is the old column style alignment option

File: public/app/plugins/panel/table/migrations.ts [hunk 1]
```diff
@@ -16,6 +28,194 @@ export const tableMigrationHandler = (panel: PanelModel<Options>): Partial<Optio
   return panel.options;
 };
 
+const transformsMap = {
+  timeseries_to_rows: 'seriesToRows',
+  timeseries_to_columns: 'seriesToColumns',
+  timeseries_aggregations: 'reduce',
+};
```
Comment (mckn) [inline public/app/plugins/panel/table/migrations.ts:35]: I think we might need to add `table -> merge` in this scenario.

**Review by mckn — COMMENTED** [review]
Great work on this @jackw! I think the code looks really nice and works as expected. We need to add support to handle the `table` and maybe the `annotations` mode as well.

`table` should migrate to the new one by adding a `merge` transform.

I'm not sure about the `annotations` @torkelo do you know how to migrate this one?

Skärmavbild 2021-01-14 kl  11 59 16

**Review by torkelo — APPROVED** [review]
works great!

File: public/app/plugins/panel/table/migrations.ts (hunk 1 above)
Comment (jackw) [reply public/app/plugins/panel/table/migrations.ts:36]: Thanks for spotting this @mckn. Have added it to the mapping.

**Review by torkelo — APPROVED** [review]

**Pull request merged**